packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -469,7 +471,7 @@ class Analytics implements IAnalytics {
         this.errorHandler.onError(err, ANALYTICS_CORE, READY_CALLBACK_INVOKE_ERROR);
       }
     } else {
-      state.eventBuffer.readyCallbacksArray.value.push(callback);
+      state.eventBuffer.readyCallbacksArray.value = [callback];
```

Comment:
you need to create a new array that appends the item I believe. 
Now you restrict that only one on ready callback can be provided by customer

suggestion:

```
state.eventBuffer.readyCallbacksArray.value = [
  ...state.eventBuffer.readyCallbacksArray.value,
  callback
]
```

Reply:
Thank you for pointing it out.

Reply:
I went thorugh all the existing codebase and we do not do any .push in signals anywhere else also